webapp/src/components/TodoList.vue

```diff
@@ -0,0 +1,215 @@
+    <div id="todo_list_head">
+      <div id="todo_list_title">Todo List</div>
+      <div class="head_right">
+        <button id="addTodo" class="bigGreen" @click="addEntry">
+          Add New Todo
+        </button>
+        <div style="font-style: italic">
+          Current editor is {{ user.username }}
+        </div>
+        <!--        <button class="smallGray" @click="editAuthor">Edit</button>-->
+      </div>
+      <!--      <div v-else class="head_right">-->
+      <!--        <button class="bigGreen" @click="saveAuthor">Save Author</button>-->
+      <!--        <input v-model="author.name" />-->
+      <!--      </div>-->
+    </div>
+    <ol>
+      <TodoItem
+        v-for="todo in this.todoListData"
+        :todo="todo"
+        :edit-mode="initialEditMode"
+        :key="todo.id"
+        @delete-entry="deleteEntry"
+        @save-entry="saveEntry"
+        @remove-entry-local="removeEntryLocal"
+      />
+    </ol>
+  </div>
+</template>
+
+<script>
+import gql from "graphql-tag";
+import TodoItem from "./TodoItem";
+
+export default {
+  name: "TodoList",
+  components: { TodoItem },
+  props: ["user"],
+  apollo: {
+    todoListData: {
+      query: gql`
+        query {
+          todoListData: todos {
+            id
+            text
+            author {
+              name
+            }
+          }
+        }
+      `
+    }
+  },
+  data() {
+    return {
+      initialEditMode: false,
+      isEditingAuthor: false
+    };
+  },
+  methods: {
+    removeEntryLocal() {
+      this.todoListData.splice(this.todoListData.length - 1, 1);
+    },
+    addEntry() {
+      this.initialEditMode = true;
+      const newTodo = {
+        message: "New Todo",
+        author: {
+          name: this.user.username
+        }
+      };
+      this.todoListData.push(newTodo);
+    },
+    deleteEntry: function(todo) {
+      this.$apollo
+        .mutate({
+          mutation: gql`
+            mutation delToDo($id: ID) {
+              delToDo(id: $id) {
+                id
+                text
+                author {
+                  name
+                }
+              }
+            }
+          `,
+          variables: {
+            id: todo.id
+          }
+        })
+        .then(body => {
+          this.todoListData = body.data.delToDo.slice();
+        });
+    },
+    saveEntry: function(todo) {
+      if (todo.id == null) {
+        this.initialEditMode = false;
+        this.$apollo
+          .mutate({
+            mutation: gql`
+              mutation addToDo($text: String!, $authorName: String!) {
+                addToDo(text: $text, authorName: $authorName) {
+                  id
+                  text
+                  author {
+                    name
+                  }
+                }
+              }
+            `,
+            variables: {
+              text: todo.text,
+              authorName: this.user.username
+            }
+          })
+          .then(body => {
```

Comment:
I would recommend `async/await` syntax.

Reply:
can the Methods from Tue component be ```async```??

Reply:
Sure, I think your build tool will take care of it.

Reply:
ok it looks a bit tidier